test/tests/functional/pbs_qstat_formats.py

```diff
@@ -104,6 +104,49 @@ def parse_json(self, dictitems, qstat_attr):
                         self.parse_json(val, qstat_attr)
         return qstat_attr
 
+    def get_qstat_attribs(self, obj_type):
+        """
+        Common function to get the qstat attributes in default format.
+        Attributes returned by this function are used to validate the
+        '-F json' format output.
+        The dictionary of attributes as returned by status() can not
+        be used directly because some attributes are printed differently
+        in '-F json' format.  Hence this function returns a modified
+        attributes list.
+        obj_type: Can be SERVER, QUEUE or JOB for qstat -Bf, qstat -Qf
+              and qstat -f respectively
+        """
+        attrs = self.server.status(obj_type)
+        qstat_attrs = []
+
+        for key, val in attrs[0].iteritems():
+            # qstat -F json output does not
+            # print the 'id' attribute. Its value
+            # is printed instead.
+            if key is 'id':
+                qstat_attrs.append(str(val))
+            else:
+                # Attributes like default_chunk.ncpus in
+                # the default format are represented as
+                # "default_chunk":{
+                #      "ncpus":1
+                #  } in -F json format.
+                # Extract the keys so they can be matched
+                # with -F json format.
+
+                k = key.split('.')
+                if k[0] not in qstat_attrs:
+                    qstat_attrs.append(str(k[0]))
+                if len(k) == 2:
+                    qstat_attrs.append(str(k[1]))
+
+            # Items in the 'Variable_List' attribute are also
+            # keys in '-F json' format. Extract them.
```

Comment:
This is true for Resource_List as well, is it not? In that case, you should updated the comment to indicate any attribute containing a list.

Reply:
I'm looking at the qstat json output of a running job, and I see three places where a comma separated list is embeded.

"resources_used":{
                "cpupercent":0,
                "cput":"00:00:00",
                "mem":"352kb",
                "ncpus":1,
                "vmem":"4360kb",
                "walltime":"00:00:08"
            }

"Resource_List":{
                "ncpus":1,
                "nodect":1,
                "place":"pack",
                "select":"1:ncpus=1"
            }

"Variable_List":{
                "PBS_O_HOME":"/home/mike",
                "PBS_O_LANG":"en_US.UTF-8",
                "PBS_O_LOGNAME":"mike",
                "PBS_O_PATH":"/usr/lib64/qt-3.3/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/opt/openmpi/bin:/opt/pbs/bin:/home/mike/.local/bin:/home/mike/bin",
                "PBS_O_MAIL":"/var/spool/mail/mike",
                "PBS_O_SHELL":"/bin/bash",
                "PBS_O_WORKDIR":"/home/mike",
                "PBS_O_SYSTEM":"Linux",
                "PBS_O_QUEUE":"workq",
                "PBS_O_HOST":"pbs-server.elusive.name"
            }

My point is that the comment applies to more than just the Variable_List attribute and should probably be reworded. Apologies if I'm missing something here.

Reply:
@mike0042 , I have added appropriate comments. Sorry that the earlier comments were confusing. Please take a look.

Reply:
@latha-subramanian : My confusion surrounds my observation that your comment only applies to Variable_List, but there are two additional sections that the code appears to apply to. Namely, resources_used and Resource_List. I think they should also be called out in the comment if you're going to mention Variable_List. Otherwise, you could make a generic comment that applies to all three sections without naming any of them. Does the code really apply ONLY to Variable_List and I'm somehow reading it wrong?

Reply:
Yes, the code applies to only Variable_List. In the default qstat -f output, only 'Variable_List' has items separated by commas. Resource_List and Resources_used are printed as separate lines like,   
    Resource_List.ncpus = 1
    Resource_List.nodect = 1
    Resource_List.place = pack

Reply:
My mistake then. Thank you for the explanation!